scipy/signal/ltisys.py

```diff
@@ -2494,3 +2496,229 @@ def place_poles(A, B, poles, method="YT", rtol=1e-3, maxiter=30):
+        the linear system :math:`AX+BU`. If ``control_index==A.shape[0]`` then
+        the linear system is controllable. Control_index is also the rank of 
+        :math:`\bar A`.
+        
+    Notes
+    -----
+    The algorithm comes from [1]_. The output from this function has been tested
+    with random controllable and uncontrollable system against Octave which
+    uses tb01ud from Slicot and all results were similar using the default
+    tolerances of `allcose`.
+    
+    References
+    ----------
+    .. [1] Datta Biswa, "Numerical Methods for Linear Control Systems",
+           Elsevier, pp 173-183, 2003.     
+
+    """
+
+    stop = False
+    P = None
+    # H_sub => the submatrix of H we are working on
+    H_sub = B
+    # where to seek the submatrices H_sub  
+    H_start = 0    
+    control_index = 0
+    
+    while not stop:
+        # page 174 step 0
+        P_update, R, _ = s_qr(H_sub, pivoting=True)   
+        # be consistent with [1] they use the form Q A P = R
+        # whereas scipy uses A P = Q R. The variables names
+        # are chosen to match [1] therefore P_update refers to "Q" from
+        # the QR decomposition and not the permutation matrix usually
+        # named "P".
+        P_update = P_update.T  
+
+        H_start = control_index
+        
+        # find the first occurence of a zero line in R, this will
+        # reveal the rank of H_sub
+        zero_lines_idx = np.where(~R.any(axis=1))[0]
+        if not zero_lines_idx.shape[0]:
+            control_index += np.min(R.shape)
+        else:
+            control_index += zero_lines_idx[0]
+            
+#        It looks like this condition from the paper is useless as the checks
+#        at the end of the loop will break anyway. If a bug is discovered the
+#        first place to look at should be this one.
+#        if control_index == A.shape[0] and H_start != 0:
+#             "if n1 + n2 + ... = n stop" page 175
+#            break 
+            
+#        form P2 = ... page 174 note that for P1 we have H_start=0 so that's
+#        OK even if it is useless   
+        P_update_full = np.zeros(
+            (H_start+P_update.shape[0],H_start+P_update.shape[1])
+            )
+        P_update_full[:H_start,:H_start] = np.eye(H_start)
+        P_update_full[H_start:,H_start:] = P_update
+
+        if P is None:  # P1
+            P = P_update_full
+        else:
+            P = np.dot(P_update_full,P)  # "update P as above page 175"
+
+        # update H and extract the new "H_k_k-1" we will work on
+        H = np.dot(np.dot(P,A),P.T)
+        H_sub = H[control_index:,H_start:control_index]
+            
+        if np.allclose(H_sub,0):  # or H_k_k-1" is a zero matrix         
+            stop = True
+            
+    Bbar = np.dot(P,B)
+    if C is None:
+        Cbar = None
+    else:
+        Cbar = np.dot(C,P.T)
+    return H,Bbar,Cbar,P,control_index
+    
```

Comment:
The issue is that there's a lot of old (even pre-PEP8) code that doesn't comply to all PEP8 rules. And we don't want to change all that at once, because (a) it's a lot of work, and (b) it ruins the code history. So there are a number of PEP8 errors excluded from the check in https://github.com/scipy/scipy/blob/master/tox.ini#L41. If you run ``pep8`` locally from the scipy repo root, it'll pick up those excludes. 

It's actually quite difficult to get ``pep8`` to ignore ``tox.ini``, but if I delele it then ``pep8 ltisys.py`` gives a lot of errors. A better alternative is ``flake8``, that doesn't use ``tox.ini`` it looks like. So try ``flake8 ltisys.py`` to see what errors you added (it's not clean on master, but much shorter than on this PR).